lambda-alb/src/AWS/Lambda/Header.hs

```diff
@@ -1,33 +1,37 @@
 module AWS.Lambda.Header where
 
 import AWS.Lambda.Runtime.Prelude
+import Data.ByteString (ByteString)
 import Data.HashMap.Strict (HashMap)
 
 import qualified Data.Aeson           as JSON
 import qualified Data.Aeson.Types     as JSON
 import qualified Data.CaseInsensitive as CI
 import qualified Data.HashMap.Strict  as HashMap
 import qualified Data.Text.Encoding   as Text
-import qualified Network.HTTP.Types   as HTTP
 
-newtype Headers = Headers [HTTP.Header]
+type HeaderName = CI.CI Text
+
+type Header = (HeaderName, ByteString)
+
+newtype Headers = Headers [Header]
   deriving stock (Eq, Show)
 
 instance JSON.ToJSON Headers where
   toJSON (Headers list)
     = JSON.Object . HashMap.fromList
-    $ bimap (decodeUtf8 . CI.original) (JSON.String . decodeUtf8) <$> list
+    $ bimap CI.original (JSON.String . decodeUtf8) <$> list
```

Comment:
@epicallan With: https://github.com/mbj/mhs/pull/94/commits/67982a5ce56075dd1bf5ad7ea9e4172e4b15d55b#r792298712 we'd get a rid of the other `decodeUtf8` which is a partial function.